File: typescript_sdk/packages/reality-capture/src/service/files.ts
import { z } from "zod";

export enum FileType {
  PRESET = "Preset",
}

export const FileSchema = z.object({
    id: z.string().describe("Id of the file"),
    name: z.string().min(3).max(256).describe("Display name of the file"),
    type: z.nativeEnum(FileType).describe("File type"),
    description: z.string().optional().describe("Description of the file"),
    deprecated: z.boolean().optional().describe("If true, this file won't be available in a long term future."),
});
export type File = z.infer<typeof FileSchema>;

export const FilesSchema = z.object({
    files: z.array(FileSchema).describe("List of files"),
});
export type Files = z.infer<typeof FilesSchema>;

File: python_sdk/src/reality_capture/service/files.py
from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class FileType(Enum):
    PRESET = "Preset"


class File(BaseModel):
    id: str = Field(description="Id of the file")
    name: str = Field(description="Display name of the file", min_length=3, max_length=256)
    type: FileType = Field(description="File type")
    description: Optional[str] = Field(None, description="Description of the file")
    deprecated: Optional[bool] = Field(None, description="If true, this file won't be available in a long term future.")


class Files(BaseModel):
    files: list[File] = Field(description="List of files")
